File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a, b):
    return [i for i in range(a, b+1)]  # Create list of ints from a to b

def applicatorFunc(inpFunc, s, a, b):
    lst = inpFunc(a, b)
    if s == 's':
        return sum(lst)
    elif s == 'a':
        return sum(lst) / len(lst)
    else:
        raise ValueError("Invalid operation")

a = 1
b = 10
print(applicatorFunc(listFunc, 'a', a, b))  # Call applicatorFunc with listFunc, 'a', a, and b as args

File: lab2-functionalprimer/javascript/func_as_arg.js
// Create list of ints from a to b
function arrFunc(a, b) {
    let arr = [];
    for (let i = a; i <= b; i++) {
        arr.push(i);
    }
    return arr;    
}

function applicatorFunc(inpFunc, s, a, b) {
    const arr = inpFunc(a, b);
    if (s == 's') {
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    } else if (s == 'a') {
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum / arr.length;
    } else {
        throw new Error("Invalid operation");
    }
}

let a = 1;
let b = 10;
let x = applicatorFunc(arrFunc, 'a', a, b); // Call applicatorFunc with arrFunc, 'a', a, and b as args
console.log(x);